app/views/login/__init__.py

```diff
@@ -37,15 +37,17 @@ def load_user_from_request(request):
     if not token:
```

Comment:
We can check here if the token is valid:
```suggestion
    if not token or token.room is None or token.logins_left == 0:
```